src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -377,21 +377,27 @@ TransactionManager::~TransactionManager() {}
 
 void TransactionManager::register_metrics()
 {
-  namespace sm = seastar::metrics;
-  metrics.add_group("tm", {
-    sm::make_counter("extents_retired_total", stats.extents_retired_total,
-		     sm::description("total number of retired extents in TransactionManager")),
-    sm::make_counter("extents_retired_bytes", stats.extents_retired_bytes,
-		     sm::description("total size of retired extents in TransactionManager")),
-    sm::make_counter("extents_mutated_total", stats.extents_mutated_total,
-		     sm::description("total number of mutated extents in TransactionManager")),
-    sm::make_counter("extents_mutated_bytes", stats.extents_mutated_bytes,
-		     sm::description("total size of mutated extents in TransactionManager")),
-    sm::make_counter("extents_allocated_total", stats.extents_allocated_total,
-		     sm::description("total number of allocated extents in TransactionManager")),
-    sm::make_counter("extents_allocated_bytes", stats.extents_allocated_bytes,
-		     sm::description("total size of allocated extents in TransactionManager")),
-  });
+  LOG_PREFIX(TransactionManager::register_metrics);
```

Comment:
i'd avoid this. as i think `register_metrics()` should never fail in a proper application. if we have to guard the code which *might* throw with the try-catch block for printing out the exception, that actually imply a bigger requirement -- we need to be able to print exception backtrace in when application start-up.

Reply:
agree.

Reply:
But the problem is that we currently (maybe also in the future) cannot guarantee to print out proper backtraces for an unhandled exception. Any logic that indirectly forwards the exception will make the backtrace unreadable. So I think we need to catch and abort undesired exceptions as early as possible. In this case, if we don't allow the exception to be propagated, the backtrace could read:
```
[Backtrace #0]
__interceptor_tcgetattr at ??:?
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /root/ceph/src/seastar/include/seastar/util/backtrace.hh:59
seastar::backtrace_buffer::append_backtrace() at /root/ceph/src/seastar/src/core/reactor.cc:766
seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at /root/ceph/src/seastar/src/core/reactor.cc:796
seastar::print_with_backtrace(char const*, bool) at /root/ceph/src/seastar/src/core/reactor.cc:808
seastar::sigabrt_action() at /root/ceph/src/seastar/src/core/reactor.cc:3570
seastar::install_oneshot_signal_handler<6, &seastar::sigabrt_action>()::{lambda(int, siginfo_t*, void*)#1}::operator()(int, siginfo_t*, void*) const at /root/ceph/src/seastar/src/core/reactor.cc:3552
seastar::install_oneshot_signal_handler<6, &seastar::sigabrt_action>()::{lambda(int, siginfo_t*, void*)#1}::_FUN(int, siginfo_t*, void*) at /root/ceph/src/seastar/src/core/reactor.cc:3548
?? ??:0
gsignal at ??:?
abort at ??:?
__cxa_throw_bad_array_new_length at ??:?
std::rethrow_exception(std::__exception_ptr::exception_ptr) at ??:?
std::terminate() at ??:?
std::rethrow_exception(std::__exception_ptr::exception_ptr) at ??:?
std::__basic_future<void>::_M_get_result() const at /usr/include/c++/9/future:719
std::future<void>::get() at /usr/include/c++/9/future:882
void SeastarRunner::run<main::{lambda()#2}>(main::{lambda()#2}&&) at /root/ceph/src/test/crimson/seastar_runner.h:67
main at /root/ceph/src/crimson/tools/store_nbd/store-nbd.cc:305 (discriminator 2)
__libc_start_main at ??:?
_start at ??:?
```

I think this is not meaningful for us to debug.

Reply:
so, we need to have a machinery which just crash the application instead of throwing an exception. probably we could leverage `seastar::on_internal_error()` and/or `seastar::on_internal_error_noexcept()` ?

Reply:
But this commit is already **the machinery** to crash the application immediately using `ceph_abort()` instead of throwing undesired exceptions into crimson logic.

Seems `seastar::on_internal_error()` is only useful when we need to control whether `on_internal_error()` aborts or throws. But we don't, as we always want the abort behavior, not throw, moreover, throwing is the default behavior. So `ceph_abort()` should be simpler and sufficient, is it correct?

Reply:
I don't really want to be sprinkling try/catch around the code base simply to print a backtrace.  I'd rather try to fix the problem that we're not getting useful backtraces.

Reply:
I'm wondering that there are limited external interfaces that may throw exceptions into our logic.

If we need to propagate undesired exceptions thoughtout the stacks in order to print a backtrace, we might instead need to aware:
1. Unexpected stack unwinding may cause destructor called incorrectly that leads to nested issues;
2. We need to retrieve the original stack traces whenever there is an indirect re-throw, like `future.get().` in this example.